src/abt-io.c

```diff
@@ -6,16 +6,19 @@
 
 #include "abt-io-config.h"
 
-#define _GNU_SOURCE
```

Comment:
Agree that it is likely for the best to remove this for portability purposes (defining this grants access to GNU/Linux specific features that aren't actually in POSIX), but building on Linux without it set produces:

```
../src/abt-io.c: In function 'abt_io_mkostemp_fn':
../src/abt-io.c:998:19: warning: implicit declaration of function 'mkostemp'; did you mean 'mkstemp'? [-Wimplicit-function-declaration]
  998 |     *state->ret = mkostemp(state->tpl, state->flags);
      |                   ^~~~~~~~
      |                   mkstemp
../src/abt-io.c: In function 'abt_io_fallocate_fn':
../src/abt-io.c:1409:19: warning: implicit declaration of function 'fallocate'; did you mean 'alloca'? [-Wimplicit-function-declaration]
 1409 |     *state->ret = fallocate(state->fd, state->mode, state->offset, state->len);
      |                   ^~~~~~~~~
      |                   alloca
```

(in other words we lose the definitions for mkostemp and fallocate).  We could get by without mkostemp, but fallocate would be nice to have on platforms that support it.  Maybe we need new autoconf tests for that?

Reply:
maybe the thing to do is to have the fallocate autoconf test add `-D_GNU_SOURCE` to the CFLAGS right there in the configure script  (rather than defining it in the .c file) since the `AC_TRY_LINK` test code snippit is using it.

Reply:
I tried that approach in https://github.com/mochi-hpc/mochi-abt-io/pull/31/commits/510271ac2d77fb9e20f9a428ad5f9130fc36fb90 if you want to confirm that it is also clean on MacOS.  It looks good on my machine.